catalog/Dockerfile

```diff
@@ -11,6 +11,8 @@ COPY build /usr/share/nginx/html
 # Copy config file
 COPY config.json.tmpl config.json.tmpl
 
-# Substitute environment variables into config.js before starting nginx.
+# Substitute environment variables into config.json and generate config.js based on that before starting nginx.
+CMD envsubst < config.json.tmpl > /usr/share/nginx/html/config.json
+CMD echo "window.QUILT_CATALOG_CONFIG = `cat /usr/share/nginx/html/config.json`" > /usr/share/nginx/html/config.js
```

Comment:
Just do `RUN cmd1 && cmd2 && cmd3` to run multiple commands.

If it doesn't work or gets annoying, you can create a script like `run_nging.sh` that does all the work. (I was trying to use `$(<...)` instead of `cat` - but either it's not supported by `/bin/sh`, or somehow `$` doesn't work - but it might be better to move everything into a helper script.)